torch/ao/quantization/quantize.py

```diff
@@ -317,69 +316,77 @@ def quantize(model, run_fn, run_args, mapping=None, inplace=False):
     return model
 
 def quantize_dynamic(model, qconfig_spec=None, dtype=torch.qint8,
-                     mapping=None, inplace=False):
-    r"""Converts a float model to dynamic (i.e. weights-only) quantized model.
-
+                     mapping=None, inplace=False, version=1):
```

Comment:
it's a bit confusing to have versions on both the APIs and the qconfig which do the same thing.  Would it make sense to reduce it to just one of these objects having a version, and the other reading that version?

Reply:
The issue is that both are functions having no state. quantize_dynamic is the user facing API and it now has a version argument.  get_default_dynamic_qconfig is a helper function that returns the recommended qconfig, and can be called independently.  Do you have a suggestion on how we can eliminate version argument for one of the functions?

Reply:
One option is to remove the get_default_dynamic_qconfig function all together. It is a new function added as part of this PR and we could have the same logic occur inside quantize_dynamic. Let me know if you prefer that option.

Reply:
IMO `QConfig` is the place which defines which observers should be used, so the version for defining the "best qconfig" should go either on the `QConfig` object itself or on utility functions which return `QConfig` objects.

Reply:
The problem is that users do not specify a qconfig for quantize_dynamic, they call it with a dtype or with no args. In this case, to preserve backward compatibility, we need a version argument to this function.  If we choose to have a get_default_Dynamic_qconfig function, then that needs a version argument too as it is the one returning the qconfig to be used.

Reply:
agreed.  As written, `qconfig_dynamic` doesn't follow what the rest of the quantization codebase is doing as it combines specifying qconfigs into the body of the quantization function.  It might be worth aligning how to best fix this.  Some solutions could be:
1. freeze the current version of `quantize_dynamic` to keep BC, and add a new version of this function which does not specify qconfigs
2. break BC and update `quantize_dynamic` to not specify qconfigs
3. maintain the current state of things (currently what this PR is doing).  This is easiest in the short term but it also not the cleanest.

Reply:
Sounds good, will follow up.